### PR TITLE
TF 2.x: Support for keras to estimator

### PR DESCRIPTION
**Pull request by vandanavk**
### Description of changes:
This PR makes a few fixes to enable keras to estimator in TF 2

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: smdebug/tensorflow/collection.py [hunk 1]
```diff
@@ -148,9 +148,6 @@ def __init__(self, collections=None, create_default=True):
                 self.create_collection(n)
             if is_tf_version_2x() and tf.executing_eagerly():
                 self.get(CollectionKeys.BIASES).include("^(?!gradient).*bias")
-                self.get(CollectionKeys.WEIGHTS).include("^weights/.*/((?!bias).)*$")
-                self.get(CollectionKeys.LOSSES).include(".*loss.*")
-                self.get(CollectionKeys.GRADIENTS).include("^gradient")
             else:
                 self.get(CollectionKeys.BIASES).include("bias")
```
Comment (vandanavk) [inline smdebug/tensorflow/collection.py:152]: why weren't loss and othernregexes set for SessionHook ?

File: docs/tensorflow.md [hunk 2]
```diff
@@ -18,9 +18,14 @@
 - This library itself supports the following versions when you use our API which requires a few minimal changes to your training script: TensorFlow 1.14, 1.15, 2.0.1, 2.1.0. Keras 2.3.
```
Comment (vandanavk) [inline docs/tensorflow.md:18]: maybe we just say we support 2.0+ versions

File: smdebug/tensorflow/collection.py (hunk 1 above)
Comment (rahul003) [reply smdebug/tensorflow/collection.py:152]: Because SessionHook doesn't identify these tensors based on regex patterns. They come from global collections maintained by TF. (Not gradients where we manually add them)

Comment (rahul003) [reply smdebug/tensorflow/collection.py:152]: Why are the above being removed btw?

File: tests/zero_code_change/tf_utils.py [hunk 3]
```diff
@@ -232,6 +231,8 @@ def neural_net(x):
 
 
 def get_data() -> "tf.contrib.learn.python.learn.datasets.base.Datasets":
+    from tensorflow.examples.tutorials.mnist import input_data
+
```
Comment (NihalHarish) [inline tests/zero_code_change/tf_utils.py:235]: Why?

File: smdebug/tensorflow/keras.py [hunk 4]
```diff
@@ -716,6 +716,11 @@ def run(*args, **kwargs):
                 # at this point we need all collections to be ready
                 # this may not be the case at creation of hook
                 # as user's code after hook might add collections
+                self.collection_manager.get(CollectionKeys.WEIGHTS).include(
+                    "^weights/.*/((?!bias).)*$"
+                )
+                self.collection_manager.get(CollectionKeys.LOSSES).include(".*loss.*")
+                self.collection_manager.get(CollectionKeys.GRADIENTS).include("^gradient")
```
Comment (NihalHarish) [inline smdebug/tensorflow/keras.py:723]: It doesn't seem like we have tested the regex patterns themselves.

File: smdebug/tensorflow/collection.py (hunk 1 above)
Comment (vandanavk) [reply smdebug/tensorflow/collection.py:152]: If I have loss regex here, then https://github.com/awslabs/sagemaker-debugger/blob/master/smdebug/tensorflow/session.py#L206 starts adding other operations in the graph with the regex "*loss*" into the loss collection. It gave 105 loss tensors in the trial instead of 1.

since these regex patterns were added for gradient tape support, moved them to grad tape related function so that it doesn't affect other hooks

File: tests/zero_code_change/tf_utils.py (hunk 3 above)
Comment (vandanavk) [reply tests/zero_code_change/tf_utils.py:235]: its causing an error with TF 2 because I'm importing tf_utils in test_estimator.py.

File: smdebug/tensorflow/keras.py (hunk 4 above)
Comment (vandanavk) [reply smdebug/tensorflow/keras.py:723]: copied them from pt regex. do we have tests for regex?

File: tests/zero_code_change/tf_utils.py (hunk 3 above)
Comment (NihalHarish) [reply tests/zero_code_change/tf_utils.py:235]: Is there a function in this file that is exclusively used with TF 2?
Then we can simply move it to a new utils file, so that we do not have this import clash.

Comment (vandanavk) [reply tests/zero_code_change/tf_utils.py:235]: there's nothing specific to TF 2. whatever is used in the TF 2 works for both TF 1 and TF 2. only this get data function is specific to TF 1
